mala/datahandling/data_shuffler.py

```diff
@@ -0,0 +1,138 @@
+import os
+
+import numpy as np
+
+from mala.common.parameters import ParametersData, Parameters
+from mala.common.parallelizer import printout
+from mala.datahandling.data_handler_base import DataHandlerBase
+
+
+class DataShuffler(DataHandlerBase):
+
+    def __init__(self, parameters: Parameters, target_calculator=None,
+                 descriptor_calculator=None):
+        super(DataShuffler, self).__init__(parameters,
+                                           target_calculator=target_calculator,
+                                           descriptor_calculator=
+                                           descriptor_calculator)
+        if self.descriptor_calculator.parameters.descriptors_contain_xyz:
+            printout("Disabling XYZ-cutting from descriptor data for "
+                     "shuffling. If needed, please re-enable afterwards.")
+            self.descriptor_calculator.parameters.descriptors_contain_xyz = \
+                False
+
+    def add_snapshot(self, input_npy_file, input_npy_directory,
+                     output_npy_file, output_npy_directory,
+                     add_snapshot_as="te",
+                     output_units="1/(eV*A^3)", input_units="None",
+                     calculation_output_file="", snapshot_type="numpy"):
+        super(DataShuffler, self).\
+            add_snapshot(input_npy_file, input_npy_directory,
+                         output_npy_file, output_npy_directory,
+                         add_snapshot_as="te",
+                         output_units=output_units, input_units=input_units,
+                         calculation_output_file=calculation_output_file,
+                         snapshot_type=snapshot_type)
+
+    def shuffle_snapshots(self, save_path, save_name="mala_shuffled_snapshot*",
```

Comment:
That is a great idea! I will implement this. 
I am not sure where to best save the seed. It definitely should be part of the parameters object, since this object should be used to document any runs for reproducibility. File names are tricky imho because they can easily be renamed. Best would be metadata, but for this we would have to use OpenPMD - which we should anyway. I will try to make this PR work with OpenPMD as well to this effect.

Reply:
Oh yes, good point, that makes the most sense.

Reply:
I have enabled OpenPMD saving. The reading is something I will have to further work on with Franz, but at any rate the seed can now be stored without issue.